Content.Server/Storage/Components/EntityStorageComponent.cs

```diff
@@ -183,8 +183,8 @@ public virtual bool CanOpen(EntityUid user, bool silent = false)
         {
             if (IsWeldedShut)
             {
-                if (!user.IsInContainer())
-                    if (!silent) Owner.PopupMessage(user, Loc.GetString("entity-storage-component-welded-shut-message"));
+                if (!user.IsInContainer() && !silent)
```

Comment:
```suggestion
                if (!silent && !user.IsInContainer())
```
So that the less complex check is done first to exit early.